src/chrome/crdpMultiplexing/crdpMultiplexor.ts

```diff
@@ -0,0 +1,201 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { LikeSocket } from 'noice-json-rpc';
+import { logger } from 'vscode-debugadapter';
+
+/* Assumptions made to implement this multiplexor:
+    1. The message IDs of CRDP don't need to be sent in order
+    2. The Domain.enable messages don't have any side-effects (We might send them multiple times)
+    3. The clients are ready to recieve domain messages when they send the Domain.enable message (A better design would be to assume that they are ready after we've sent the response for that message, but this approach seems to be working so far)
+    4. The clients never disable any Domain
+    5. The clients enable all domains in the first 60 seconds after they've connected
+ */
+
+function extractDomain(method: string): string {
+    const methodParts = method.split(".");
+    if (methodParts.length === 2) {
+        return methodParts[0];
+    } else {
+        throwCriticalError(`The method ${method} didn't have exactly two parts`);
+        return "Unknown";
+    }
+}
+
+function encodifyChannel(channelId: number, id: number): number {
+    return id * 10 + channelId;
```

Comment:
I like this approach because without any extra effort we can easily tell which channel sent each message...